### PR TITLE
[AIRFLOW-5819] Update AWSBatchOperator default value

### PR DESCRIPTION
**Pull request by domantasjurkus**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-5819) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

The default value `None` throws an error, should be changed to `{}`.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

I am actually not sure why the issue was not captured in the existing test:
https://github.com/apache/airflow/blob/73bf718358c01cc41e5f23b914a8824a2665a28c/tests/contrib/operators/test_awsbatch_operator.py#L39

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/contrib/operators/awsbatch_operator.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ class AWSBatchOperator(BaseOperator):
     template_fields = ('job_name', 'overrides',)
 
     @apply_defaults
-    def __init__(self, job_name, job_definition, job_queue, overrides, array_properties=None,
+    def __init__(self, job_name, job_definition, job_queue, overrides, array_properties={},
```
Comment (ashb) [inline airflow/contrib/operators/awsbatch_operator.py]: You shouldn't do it like this. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Instead in the method do

```
        self.array_properties = array_properties if array_properties is not None else {}
```

**Review by ashb — CHANGES_REQUESTED** [review]

File: airflow/contrib/operators/awsbatch_operator.py (hunk 1 above)
Comment (domantasjurkus) [reply airflow/contrib/operators/awsbatch_operator.py]: Thanks for pointing out, updated.

**Comment by kaxil** [comment]
Can you fix the failing tests too:

```
======================================================================
7) FAIL: test_execute_with_failures (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/contrib/operators/test_awsbatch_operator.py line 102 in test_execute_with_failures
      arrayProperties=None
    /usr/local/lib/python3.6/unittest/mock.py line 825 in assert_called_once_with
      return self.assert_called_with(*args, **kwargs)
    /usr/local/lib/python3.6/unittest/mock.py line 814 in assert_called_with
      raise AssertionError(_error_message()) from cause
   AssertionError: Expected call: submit_job(arrayProperties=None, containerOverrides={}, jobDefinition='hello-world', jobName='51455483-c62c-48ac-9b88-53a6a725baa3', jobQueue='queue')
   Actual call: submit_job(arrayProperties={}, containerOverrides={}, jobDefinition='hello-world', jobName='51455483-c62c-48ac-9b88-53a6a725baa3', jobQueue='queue')
   -------------------- >> begin captured logging << --------------------
   airflow.task.operators: INFO: Running AWS Batch Job - Job definition: hello-world - on queue queue
   airflow.task.operators: INFO: AWSBatchOperator overrides: {}
   airflow.task.operators: INFO: AWS Batch Job started: 
   airflow.task.operators: INFO: AWS Batch Job has failed executed
   --------------------- >> end captured logging << ---------------------
======================================================================
8) FAIL: test_execute_without_failures (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator)
----------------------------------------------------------------------
   Traceback (most recent call last):
    /usr/local/lib/python3.6/unittest/mock.py line 1183 in patched
      return func(*args, **keywargs)
    tests/contrib/operators/test_awsbatch_operator.py line 81 in test_execute_without_failures
      arrayProperties=None
    /usr/local/lib/python3.6/unittest/mock.py line 825 in assert_called_once_with
      return self.assert_called_with(*args, **kwargs)
    /usr/local/lib/python3.6/unittest/mock.py line 814 in assert_called_with
      raise AssertionError(_error_message()) from cause
   AssertionError: Expected call: submit_job(arrayProperties=None, containerOverrides={}, jobDefinition='hello-world', jobName='51455483-c62c-48ac-9b88-53a6a725baa3', jobQueue='queue')
   Actual call: submit_job(arrayProperties={}, containerOverrides={}, jobDefinition='hello-world', jobName='51455483-c62c-48ac-9b88-53a6a725baa3', jobQueue='queue')
   -------------------- >> begin captured logging << --------------------
   airflow.task.operators: INFO: Running AWS Batch Job - Job definition: hello-world - on queue queue
   airflow.task.operators: INFO: AWSBatchOperator overrides: {}
   airflow.task.operators: INFO: AWS Batch Job started: {'jobName': '51455483-c62c-48ac-9b88-53a6a725baa3', 'jobId': '8ba9d676-4108-4474-9dca-8bbac1da9b19'}
   airflow.task.operators: INFO: AWS Batch Job has been successfully executed: {'jobName': '51455483-c62c-48ac-9b88-53a6a725baa3', 'jobId': '8ba9d676-4108-4474-9dca-8bbac1da9b19'}
   --------------------- >> end captured logging << ---------------------
======================================================================
9) FAIL: test_init (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/contrib/operators/test_awsbatch_operator.py line 56 in test_init
      self.assertEqual(self.batch.array_properties, None)
   AssertionError: {} != None
```

File: airflow/contrib/operators/awsbatch_operator.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def __init__(self, job_name, job_definition, job_queue, overrides, array_propert
         self.job_definition = job_definition
         self.job_queue = job_queue
         self.overrides = overrides
-        self.array_properties = array_properties
+        self.array_properties = array_properties if array_properties is not None else {}
```
Comment (feluelle) [inline airflow/contrib/operators/awsbatch_operator.py]:
```suggestion
        self.array_properties = array_properties or {}
```
Why not just like that?

Comment (feluelle) [reply airflow/contrib/operators/awsbatch_operator.py]:
```python
>>> a = None or {}
>>> a
{}
```

Comment (domantasjurkus) [reply airflow/contrib/operators/awsbatch_operator.py]: lovely

**Review by kaxil — APPROVED** [review]

**Pull request merged**